src/core/load-fragment.c

```diff
@@ -6573,3 +6575,53 @@ int config_parse_tty_size(
 
         return config_parse_unsigned(unit, filename, line, section, section_line, lvalue, ltype, rvalue, data, userdata);
 }
+
+int config_parse_log_filter_regex(
+                const char *unit,
+                const char *filename,
+                unsigned line,
+                const char *section,
+                unsigned section_line,
+                const char *lvalue,
+                int ltype,
+                const char *rvalue,
+                void *data,
+                void *userdata) {
+
+        _cleanup_(pattern_freep) pcre2_code *cs = NULL;
+        ExecContext *c = ASSERT_PTR(data);
+        const char *pattern = ASSERT_PTR(rvalue);
+        int r;
+
+        assert(filename);
+        assert(lvalue);
+
+        if (isempty(pattern)) {
+                c->log_filter_regex = mfree(c->log_filter_regex);
+                return 0;
+        }
+
+        if (pattern[0] == '~') {
+                if (pattern[1] == '\0')
+                        /* LogFilterRegex=~ is considered invalid, we keep the previous pattern. */
+                        return 0;
```

Comment:
We should `log_syntax()` here to inform the user that their pattern is considered invalid